sqlalchemy_ibmi/base.py

```diff
@@ -641,86 +641,59 @@ def fire_sequence(self, seq, type_):
+class IBMiDb2Dialect(default.DefaultDialect):
 
+    statement_compiler = DB2Compiler
+    ddl_compiler = DB2DDLCompiler
+    type_compiler = DB2TypeCompiler
+    preparer = DB2IdentifierPreparer
+    execution_ctx_cls = DB2ExecutionContext
+
+    def initialize(self, connection):
+        super().initialize(connection)
+        self.dbms_ver = getattr(connection.connection, 'dbms_ver', None)
+        self.dbms_name = getattr(connection.connection, 'dbms_name', None)
+
+    # TODO These methods are overridden from the default dialect and should be
+    #  implemented
+
+    driver = "pyodbc"
     name = 'sqlalchemy_ibmi'
     max_identifier_length = 128
     encoding = 'utf-8'
-    default_paramstyle = 'qmark'
     colspecs = COLSPECS
     ischema_names = ISCHEMA_NAMES
-    supports_unicode_binds = False
     returns_unicode_strings = False
     postfetch_lastrowid = True
     supports_native_boolean = False
     preexecute_sequences = False
     supports_alter = True
     supports_sequences = True
     sequences_optional = True
-    supports_unicode_statements = True
     supports_sane_rowcount = False
-    supports_sane_multi_rowcount = False
     # TODO Investigate if supports_native_decimal needs to be True or False
-    supports_native_decimal = False
     supports_char_length = True
-    pyodbc_driver_name = "IBM i Access ODBC Driver"
     requires_name_normalize = True
     supports_default_values = False
     supports_empty_insert = False
     two_phase_transactions = False
     savepoints = True
+    supports_sane_rowcount_returning = False
+    supports_sane_multi_rowcount = False
 
-    statement_compiler = DB2Compiler
-    ddl_compiler = DB2DDLCompiler
-    type_compiler = DB2TypeCompiler
-    preparer = DB2IdentifierPreparer
-    execution_ctx_cls = DB2ExecutionContext
-
-    def initialize(self, connection):
-        super().initialize(connection)
-        self.dbms_ver = getattr(connection.connection, 'dbms_ver', None)
-        self.dbms_name = getattr(connection.connection, 'dbms_name', None)
-
-    # TODO These methods are overridden from the default dialect and should be
-    #  implemented
-
-    def get_temp_table_names(self, connection, schema=None, **kw):
-        pass
-
-    def get_temp_view_names(self, connection, schema=None, **kw):
-        pass
-
-    def get_check_constraints(self, connection, table_name, schema=None, **kw):
-        pass
-
-    def get_table_comment(self, connection, table_name, schema=None, **kw):
-        pass
-
-    def _get_server_version_info(self, connection):
-        pass
-
-    def do_begin_twophase(self, connection, xid):
-        pass
-
-    def do_prepare_twophase(self, connection, xid):
-        pass
-
-    def do_rollback_twophase(self, connection, xid, is_prepared=True,
-                             recover=False):
-        pass
-
-    def do_commit_twophase(self, connection, xid, is_prepared=True,
-                           recover=False):
-        pass
```

Comment:
Why did these get removed?

Reply:
Removed for tests to work, but I will remove these in the testing pr rather than here, then I will implement get_check_constraints and get_table_comment in abstract-methods branch